src/hooks/usePokemonDetails.ts

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from "react-query";
+import { PokemonDetails } from "../interfaces/pokemonDetails";
+import { PokemonItem } from "../interfaces/pokemonItem";
+import { getPokemonDetails } from "../services/pokemonDetailsService";
+
+const usePokemonDetails = (pokemon: PokemonItem | null) => {
+  return useQuery<PokemonDetails | undefined, Error>(
+    ["pokemon-details", pokemon?.name],
+    () => {
+      if (pokemon) return getPokemonDetails(pokemon?.url);
```

Comment:
Seguendo il discorso "invariant" di ieri, se qui invece di fare cosi', ti metti a fare throw, il codice diventa piu' affidabile in teoria, perche' se per caso `enabled` e' `true` quando non dovrebbe, la funzione fallisce invece di fare finta di nulla :)